source/transfers/test/TransferHandlerRegistry.js

```diff
@@ -0,0 +1,1008 @@
+    })
+
+    it('Checks balances...', async () => {
+      ok(
+        await balances(aliceAddress, [
+          [tokenAST, 800],
+          [tokenDAI, 50],
+        ]),
+        'Alice balances are incorrect'
+      )
+      ok(
+        await balances(bobAddress, [
+          [tokenAST, 200],
+          [tokenDAI, 950],
+        ]),
+        'Bob balances are incorrect'
+      )
+    })
+
+    it('Checks balances...', async () => {
+      ok(
+        await balances(aliceAddress, [
+          [tokenAST, 800],
+          [tokenDAI, 50],
+        ]),
+        'Alice balances are incorrect'
+      )
+      ok(
+        await balances(bobAddress, [
+          [tokenAST, 200],
+          [tokenDAI, 950],
+        ]),
+        'Bob balances are incorrect'
+      )
+    })
+
+    it('Checks that Alice cannot trade more than approved (200 AST)', async () => {
+      const order = await orders.getOrder({
+        signer: {
+          wallet: aliceAddress,
+          token: tokenAST.address,
+          amount: 200,
+        },
+      })
+
+      await reverted(swap(order, { from: bobAddress }))
+    })
+
+    it('Checks that Bob can not trade more than he holds', async () => {
+      const order = await orders.getOrder({
+        signer: {
+          wallet: bobAddress,
+          token: tokenDAI.address,
+          amount: 1000,
+        },
+        sender: {
+          wallet: aliceAddress,
+        },
+      })
+
+      order.signature = await signatures.getWeb3Signature(
+        order,
+        bobAddress,
+        swapAddress,
+        GANACHE_PROVIDER
+      )
+
+      await reverted(swap(order, { from: aliceAddress }), 'TRANSFER_FAILED')
+    })
+
+    it('Checks remaining balances and approvals', async () => {
+      ok(
+        await balances(aliceAddress, [
+          [tokenAST, 800],
+          [tokenDAI, 50],
+        ]),
+        'Alice balances are incorrect'
+      )
+      ok(
+        await balances(bobAddress, [
+          [tokenAST, 200],
+          [tokenDAI, 950],
+        ]),
+        'Bob balances are incorrect'
+      )
+      // Alice and Bob swapped 200 AST for 50 DAI above, thereforeL
+      // Alice's 200 AST approval is now all gone
+      // Bob's 1000 DAI approval has decreased by 50
+      ok(
+        await allowances(aliceAddress, swapAddress, [
+          [tokenAST, 50],
+          [tokenDAI, 0],
+        ])
+      )
+      ok(
+        await allowances(bobAddress, swapAddress, [
+          [tokenAST, 0],
+          [tokenDAI, 950],
+        ])
+      )
+    })
+
+    it('Adding an id with fungible token will cause revert', async () => {
+      const order = await orders.getOrder({
+        signer: {
+          wallet: aliceAddress,
+          token: tokenAST.address,
+          amount: 1,
+          id: 30,
+        },
+        sender: {
+          wallet: bobAddress,
+          token: tokenDAI.address,
+          amount: 1,
+        },
+        affiliate: {
```

Comment:
remove the affiliate from this test - its added along with the ID so is confusing. The next test is the affiliate test.